sycl-fusion/passes/target/TargetFusionInfo.cpp

```diff
@@ -0,0 +1,371 @@
+//==---------------------- TargetFusionInfo.cpp ----------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "TargetFusionInfo.h"
+
+#include "llvm/ADT/Triple.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/IR/InstrTypes.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/IntrinsicsNVPTX.h"
+
+namespace llvm {
+class TargetFusionInfoImpl {
+
+public:
+  virtual ~TargetFusionInfoImpl() = default;
+
+  virtual void notifyFunctionsDelete(llvm::ArrayRef<Function *> Funcs) const {
+    (void)Funcs;
+  }
```

Comment:
```suggestion
  virtual void notifyFunctionsDelete([[maybe_unused]] llvm::ArrayRef<Function *> Funcs) const {}
```
NIT. Same for all below.